### PR TITLE
review bank 

### PR DESCRIPTION
**Pull request by imbenrabi**

### CODE REVIEW

File: server/models/user.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    next();
+})
+
+/**delete user tasks when user is removed */
+userSchema.pre('remove', async function (next) {
+    const user = this;
+    await Task.deleteMany({ owner: user._id })
+    next();
+})
+
```
Comment (YonatanMateh) [inline server/models/user.js:116]: Nice job with all the authentication

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+      pageTitle: 'My Transactions'
+    }
+  }
+
+  setData = async (token, del = false) => {
+    try {
+      const config = {
+        headers: { Authorization: `Bearer ${token}` }
+      };
+      const resp = await axios.get('/transactions', config);
```
Comment (YonatanMateh) [inline src/App.js:42]: Will be a good idea (as a continuation to the server architecture), to extract all API calls to a service (something like: `HttpServicez)

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+  getTransactionsData = async () => {
+    try {
+      const token = authService.getToken()
+      const config = {
+        headers: { Authorization: `Bearer ${token}` }
+      };
+
+      const resp = await axios.get('/transactions', config);
+      const transactions = resp.data.data;
+      const catNames = this.getCategoryNames(transactions);
```
Comment (YonatanMateh) [inline src/App.js:62]: see the note above.
move all logic outside the component, into a service

File: src/App.js [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+      return { transactions, aggregatedTransactions }
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  getAggrData = async (categoryNames) => {
+    const aggrTransactions = []
+    categoryNames.forEach(async (cat) => {
+      const aggr = await this.getAggrCategory(cat);
```
Comment (YonatanMateh) [inline src/App.js:74]: You should not use async-await with forEach

File: src/App.js [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+        return aggrCat
+      } catch (error) {
+        throw error;
+      }
+    }
+  }
+
+  renderBalance = () => {
+    let balance = 0;
+    this.state.transactions.forEach(t => balance += t.amount)
```
Comment (YonatanMateh) [inline src/App.js:120]: check [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) method...

File: src/components/TransactionForm.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            vendor: '',
+            category: ''
+        }
+    }
+
+    handleInputChange = (e) => {
+        const target = e.target;
+        const value = target.value;
+        const name = target.name;
+
```
Comment (YonatanMateh) [inline src/components/TransactionForm.js:21]: use destructuring https://hacks.mozilla.org/2015/05/es6-in-depth-destructuring/

File: src/components/TransactionForm.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        this.setState({ [name]: value });
+    }
+
+    invalidInputs(amount, vendor, category) {
+        if (!amount || !vendor || !category || amount === 0) {
+            return true;
+        }
+        return false;
```
Comment (YonatanMateh) [inline src/components/TransactionForm.js:29]: you can just return the expression `return (!amount || !vendor || !category || amount === 0)`

File: src/services/Auth.service.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+export class AuthService {
+    login(token) {
```
Comment (YonatanMateh) [inline src/services/Auth.service.js:2]: It's not an Auth service, it's LocalStorage service

File: src/services/Auth.service.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+    getToken() {
+        return localStorage.getItem('userToken');
+    }
+
+    logout() {
+        localStorage.setItem('userToken', null);
+    }
+
+    isLoggedIn() {
```
Comment (YonatanMateh) [inline src/services/Auth.service.js:15]: can be simple like this:
`isLoggedIn = () => (this.getToken() === null || this.getToken() === 'null' || !this.getToken())`

File: server/models/user.js (hunk 1 above)
Comment (imbenrabi) [reply server/models/user.js:116]: Thank you:)

File: src/App.js (hunk 2 above)
Comment (imbenrabi) [reply src/App.js:42]: It is a good idea, I'll add it to my todo list.

File: src/App.js (hunk 4 above)
Comment (imbenrabi) [reply src/App.js:74]: I see, I'll read on it today and refactor.

File: src/App.js (hunk 5 above)
Comment (imbenrabi) [reply src/App.js:120]: actually started doing it with reduce and got a little confused.
I will retry.

File: src/components/TransactionForm.js (hunk 6 above)
Comment (imbenrabi) [reply src/components/TransactionForm.js:21]: will do

**Pull request closed without merging**